signer/src/storage/in_memory.rs

```diff
@@ -414,6 +422,97 @@ impl super::DbRead for SharedStore {
             .collect())
     }
 
+    async fn get_signer_utxo(
+        &self,
+        aggregate_key: &PublicKey,
+    ) -> Result<Option<SignerUtxo>, Error> {
+        let Some(chain_tip) = self.get_bitcoin_canonical_chain_tip().await? else {
+            return Err(Error::NoChainTip);
+        };
+
+        let script_pubkey = aggregate_key.signers_script_pubkey();
+        let store = self.lock().await;
+        let bitcoin_blocks = &store.bitcoin_blocks;
+        let first = bitcoin_blocks.get(&chain_tip);
+
+        // Traverse the canonical chain backwards and find the first block containing relevant sbtc tx(s)
+        let sbtc_txs = std::iter::successors(first, |block| bitcoin_blocks.get(&block.parent_hash))
+            .filter_map(|block| {
+                store
+                    .bitcoin_block_to_transactions
+                    .get(&block.block_hash)
+                    .and_then(|txs| {
+                        let mut sbtc_txs = txs
+                            .iter()
+                            .filter_map(|tx| store.raw_transactions.get(&tx.into_bytes()))
+                            .filter(|sbtc_tx| {
+                                sbtc_tx.tx_type == model::TransactionType::SbtcTransaction
+                            })
+                            .map(|tx| {
+                                bitcoin::Transaction::consensus_decode(
+                                    &mut tx.tx.clone().as_slice(),
+                                )
+                                .unwrap()
+                            })
+                            .filter(|tx| {
+                                tx.output
+                                    .first()
+                                    .is_some_and(|out| out.script_pubkey == script_pubkey)
+                            })
+                            .peekable();
+                        if sbtc_txs.peek().is_some() {
+                            Some(sbtc_txs.collect::<Vec<_>>())
+                        } else {
+                            None
+                        }
+                    })
+            })
+            .next();
+
+        if sbtc_txs.is_none() {
+            return Ok(None);
+        }
+        // `sbtc_txs` contains all the txs in the highest canonical block where the first
+        // output is spendable by script_pubkey
+        let sbtc_txs = sbtc_txs.unwrap();
```

Comment:
You could express the same thing without an unwrap, by using the let-else pattern.